sktime/forecasting/trend.py

```diff
@@ -69,14 +74,11 @@ def fit(self, y, X=None, fh=None):
         -------
         self : returns an instance of self.
         """
-        self._is_fitted = False
 
         if X is not None:
```

Comment:
while we're here, let's just remove this error raised, see #1014

Reply:
Agreed, probably legacy code before we had agreed on how to handle exogenous data.

Reply:
Hello @fkiraly , Is there something I am suppose to resolve here? I cannot seem to see it.

Reply:
I meant, remove the `NotImplementedError` being thrown when `X` is passed. The problem is described in #1014.

This is not strictly part of the refactoring, but let's do this while we're already here.

Reply:
noted